controllers/k8ssandracluster_controller.go

```diff
@@ -106,16 +112,16 @@ func (r *K8ssandraClusterReconciler) Reconcile(ctx context.Context, req ctrl.Req
 
 			if err = remoteClient.Get(ctx, dcKey, actualDc); err == nil {
 				if actualHash, found := actualDc.Annotations[resourceHashAnnotation]; !(found && actualHash == desiredDcHash) {
-					logger.Info("Updating datacenter", "CassandraDatacenter", dcKey)
-					actualDc = actualDc.DeepCopy()
-					resourceVersion := actualDc.GetResourceVersion()
-					desiredDc.DeepCopyInto(actualDc)
-					actualDc.SetResourceVersion(resourceVersion)
-					if err = remoteClient.Update(ctx, actualDc); err != nil {
-						logger.Error(err, "Failed to update datacenter", "CassandraDatacenter", dcKey)
-						return ctrl.Result{}, err
+						logger.Info("Updating datacenter", "CassandraDatacenter", dcKey)
+						actualDc = actualDc.DeepCopy()
+						resourceVersion := actualDc.GetResourceVersion()
```

Comment:
Why is the ResourceVersion copied? I know the api-server prevents it from being set in the Create clause so I'm surprised it allows it in the Update at all.

I'm not sure it's a good way to follow changes.